src/CMakeLists.txt

```diff
@@ -63,8 +64,25 @@ if(NOT CMAKE_BUILD_TYPE)
           FORCE
   )
 endif()
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  # ThinLTO is causing build issues
+  # FullLTO is slow, but may yield performance improvements.
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -flto")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -flto")
+else()
+  # Policy that enables INTERPROCEDURAL_OPTIMIZATION
+  set(CMAKE_POLICY_DEFAULT_CMP0069 NEW)
+  cmake_policy(SET CMP0069 NEW)
+  # Clang has some issues with thinLTO (segfaults on final linking)
+  set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
+endif()
 # avoid ld visibility warnings, should be done by CMAKE_CXX_VISIBILITY_PRESET
 # but need cmake_policy(SET CMP0063 NEW) also which seems to not work
+# TODO Figure out how to remove the visibility flags from Bullet
+#set(CMAKE_POLICY_DEFAULT_CMP0063 NEW)
```

Comment:
Saving this for a later time, Figured out how to properly enable the CMake policy, but we need to work through our dependencies an make sure Bullet is built properly with old CMake versions in Shared Lib mode. Saving that for a future PR.